Terminal.Gui/Windows/FileDialog2.cs

```diff
@@ -0,0 +1,2118 @@
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using NStack;
+using Terminal.Gui.Trees;
+using static System.Environment;
+using static Terminal.Gui.OpenDialog;
+
+namespace Terminal.Gui {
+
+	/// <summary>
+	/// Modal dialog for selecting files/directories.  Has auto-complete and expandable
+	/// navigation pane (Recent, Root drives etc).
+	/// </summary>
+	public class FileDialog2 : Dialog {
+
+		// TODO : expose these somehow for localization without compromising case/switch statements
+		private const string HeaderFilename = "Filename";
+		private const string HeaderSize = "Size";
+		private const string HeaderModified = "Modified";
+		private const string HeaderType = "Type";
+
+		/// <summary>
+		/// The maximum number of results that will be collected
+		/// when searching before stopping.
+		/// </summary>
+		/// <remarks>
+		/// This prevents performance issues e.g. when searching
+		/// root of file system for a common letter (e.g. 'e').
+		/// </remarks>
+		public int MaxSearchResults {get;set;} = 10000;
```

Comment:
Should we make this configurable? If so, simply add `[SerializableConfigurationProperty (Scope = typeof(SettingsScope))]` and put the default into `./Terminal.Gui/Resources/config.json`.